chapters/classes.tex

```diff
@@ -217,17 +217,31 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
 
 In addition, instances of classes extending from \lstinline!ExternalObject! may have type prefixes \lstinline!parameter! and \lstinline!constant!, and in functions also type prefixes
 \lstinline!input! and \lstinline!output!, see \cref{external-objects}.
 
-The type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! of a structured component (except as described above) are also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct; e.g.\ the \lstinline!flow! prefix can be used on a record component and all the record elements will generate zero-sum equations, even if elements of a record shall not be declared with the \lstinline!flow! prefix).  When any of the type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes.  The corresponding rules for the type prefixes \lstinline!discrete!, \lstinline!parameter! and \lstinline!constant! are described in \cref{variability-of-structured-entities} for structured components.
+Variables declared with the \lstinline!stream! type prefix shall be a subtype of \lstinline!Real!.
+This is further restricted in \cref{definition-of-stream-connectors}.
+
+The type prefix \lstinline!flow! of a component that is not a primitive element (see \cref{primitive-elements}), is also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct).
+Primitive elements with the \lstinline!flow! type prefix shall be a subtype of \lstinline!Real! or an operator record defining an additive group, see \cref{generation-of-connection-equations}.
+
+The type prefixes \lstinline!input! and \lstinline!output! of a structured component (except as described above) are also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct).
+
+When any of the type prefixes \lstinline!flow!, \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes, nor can they have \lstinline!stream! prefix.
+The corresponding rules for the type prefixes \lstinline!discrete!, \lstinline!parameter! and \lstinline!constant! are described in \cref{variability-of-structured-entities} for structured components.
+
+\begin{nonnormative}
+The prefixes \lstinline!stream! is not included with \lstinline!flow!, \lstinline!input! and \lstinline!output!, as the corresponding rules would not be relevant based on other semantic restrictions.
```

Comment:
Should be singular:
```suggestion
The prefix \lstinline!stream! is not included with \lstinline!flow!, \lstinline!input! and \lstinline!output!, as the corresponding rules would not be relevant based on other semantic restrictions.
```
Does it still make sense to include it? I think if stream is removed from the list on line 220 as I commented above, it would not be necessary to have this sentence or the one below. It would also be easier to follow.

Also it's still not very clear where stream is not included.

Reply:
I rewrote that - if it still not clear, I would propose to just remove that entire non-normative block.

To me it makes sense to explain that stream, flow, input, output could be treated uniformly - if we ignore semantically invalid combinations; but it is just intended to help - so if it doesn't it is better to remove it.